src/main/scala/rise/core/Expr.scala

```diff
@@ -8,6 +9,19 @@ sealed abstract class Expr {
   val t: Type
   def setType(t: Type): Expr
   override def toString: String = showRise(this)
+  override def hashCode(): Int = this match {
+    case _: Identifier => 17
```

Comment:
I think that this might impact negatively all of our code which uses sets/maps of `Identifier`s? Should they map over the name instead to have a proper hash then?

Reply:
can this change also impact code such as subsitutions?